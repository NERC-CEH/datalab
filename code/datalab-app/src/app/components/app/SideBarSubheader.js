import React from 'react';
import { withStyles } from 'material-ui/styles';
import ListSubheader from 'material-ui/List/ListSubheader';

const styles = theme => ({
  subheader: {
    fontSize: 'larger',
    color: theme.palette.secondary[400],
  },
});

const SideBarSubheader = ({ classes, children }) => (
  <ListSubheader className={classes.subheader} disableSticky>{children}</ListSubheader>
);

export default withStyles(styles)(SideBarSubheader);
