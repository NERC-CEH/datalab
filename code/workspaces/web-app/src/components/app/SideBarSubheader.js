import React from 'react';
import withStyles from '@mui/styles/withStyles';
import ListSubheader from '@mui/material/ListSubheader';

const styles = theme => ({
  subheader: {
    fontSize: 'larger',
    color: theme.palette.secondary[400],
  },
});

const SideBarSubheader = ({ classes, children }) => (
  <ListSubheader className={classes.subheader} disableSticky>{children}</ListSubheader>
);

export default withStyles(styles)(SideBarSubheader);
