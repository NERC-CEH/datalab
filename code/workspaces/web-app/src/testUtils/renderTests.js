import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import theme from '../theme';

const AllTheProviders = ({ children }) => (
    <MuiThemeProvider theme={theme}>
        {children}
    </MuiThemeProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
