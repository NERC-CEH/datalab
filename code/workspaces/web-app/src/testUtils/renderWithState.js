import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import promise from 'redux-promise-middleware';
import {
  PROMISE_TYPE_PENDING,
  PROMISE_TYPE_SUCCESS,
  PROMISE_TYPE_FAILURE,
} from '../actions/actionTypes';

const promiseTypeSuffixes = [PROMISE_TYPE_PENDING, PROMISE_TYPE_SUCCESS, PROMISE_TYPE_FAILURE];

const TEST_USER = 'test-user';

export const buildDefaultTestState = () => ({
  authentication: {
    identity: {
      sub: TEST_USER,
    },
    permissions: {
      fetching: { inProgress: false },
      value: [],
    },
  },
  projectUsers: {
    fetching: { inProgress: false },
    value: [
      { userId: TEST_USER, role: 'admin' },
    ],
  },
});

export const renderWithState = (
  initialState,
  ComponentToRender,
  props = {},
  renderOptions = {},
) => {
  const mockStore = configureStore([promise({ promiseTypeSuffixes })]);
  const store = mockStore(initialState);

  const wrapper = render(
    <Provider store={store}>
      <ComponentToRender {...props} />
    </Provider>,
    renderOptions,
  );

  return {
    wrapper,
    store,
  };
};
