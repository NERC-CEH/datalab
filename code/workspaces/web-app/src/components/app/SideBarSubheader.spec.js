import React from 'react';
import { render } from '../../testUtils/renderTests';
import SideBarSubheader from './SideBarSubheader';

describe('SideBarSubheader', () => {
  it('renders correct snapshot', () => {
    const wrapper = render(<SideBarSubheader>{'expectedChild'}</SideBarSubheader>);
    expect(wrapper.container).toMatchSnapshot();
  });
});
