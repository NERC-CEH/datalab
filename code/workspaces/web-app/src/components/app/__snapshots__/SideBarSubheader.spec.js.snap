// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarSubheader renders correct snapshot 1`] = `
<div>
  <li
    class="MuiListSubheader-root MuiListSubheader-gutters SideBarSubheader-subheader-1 css-x9njn-MuiListSubheader-root"
  >
    expectedChild
  </li>
</div>
`;
